Microsoft.Toolkit.Services/Services/Twitter/TwitterDataProvider.cs

```diff
@@ -60,16 +56,54 @@ public class TwitterDataProvider : Toolkit.Services.DataProviderBase<TwitterData
         /// </summary>
         public bool LoggedIn { get; private set; }
 
+        /// <summary>
+        /// Gets the url for upload images for Twitter api.
+        /// </summary>
+        public string PublishUrl
+        {
+            get { return PUBLISHURL; }
+        }
+
+        /// <summary>
+        /// Gets the base Url for Twitter api.
+        /// </summary>
+        public string BaseUrl
+        {
+            get { return BASEURL; }
+        }
+
+        /// <summary>
+        /// Gets the tokens to execute request.
+        /// </summary>
+        public TwitterOAuthTokens Tokens
+        {
+            get { return _tokens; }
+        }
+
+        /// <summary>
+        /// Gets a reference for the SignatureManager used on the initialize.
+        /// </summary>
+        public ISignatureManager SigntureManager
+        {
+            get { return _signatureManager; }
+        }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="TwitterDataProvider"/> class.
         /// Constructor.
         /// </summary>
         /// <param name="tokens">OAuth tokens for request.</param>
-        public TwitterDataProvider(TwitterOAuthTokens tokens)
+        /// <param name="authenticationBroker">Authentication result interface.</param>
+        /// <param name="passwordManager">Platform password manager</param>
+        /// <param name="storageManager">Platform storage provider</param>
+        /// <param name="signatureManager">Platform signature manager</param>
+        public TwitterDataProvider(TwitterOAuthTokens tokens, IAuthenticationBroker authenticationBroker, IPasswordManager passwordManager, IStorageManager storageManager, ISignatureManager signatureManager)
```

Comment:
Breaking change, there should be a UWP only constructor that uses the platform specific implementations.